src/traces/scatter/plot.js

```diff
@@ -458,8 +458,10 @@ function plotOne(gd, idx, plotinfo, cdscatter, cdscatterAll, element, transition
         join.enter().append('g').classed('textpoint', true).append('text');
 
         join.each(function(d) {
-            var sel = transition(d3.select(this).select('text'));
-            Drawing.translatePoint(d, sel, xa, ya);
+            var g = d3.select(this);
+            var sel = transition(g.select('text'));
+            var hasTextPt = Drawing.translatePoint(d, sel, xa, ya);
+            if(!hasTextPt) g.remove();
```

Comment:
`Drawing.translatePoint` gets used in [line 436 above](https://github.com/plotly/plotly.js/pull/1672/files#diff-b15c528765724e5b795253c8314bb750R436) too, and could short-circuit the lines after it if it removed the node.